File: gateway/golang/routes/user_route.go
package routes

import (
	controller "gateway/controllers"
	middleware "gateway/middlewares"

	"github.com/labstack/echo/v4"
)

func UserRoute(e *echo.Echo, controller controller.UserController) {
	e.POST("/api/v1/users/register", controller.Register)
	e.POST("/api/v1/users/login", controller.Login, middleware.GetSessionId)
	e.POST("/api/v1/users/logout", controller.Logout, middleware.GetSessionId)
}

File: gateway/nodejs/src/routes/user-route.js
import express from "express";
import userController from "../controllers/user-controller.js";
import sessionIdMiddleware from "../middlewares/session-id-middleware.js";

const userRouter = () => {
    const userRouter = new express.Router()
    userRouter.post("/api/v1/users/register", userController.register)
    userRouter.post("/api/v1/users/login", sessionIdMiddleware.getSessionId, userController.login)
    userRouter.post("/api/v1/users/logout", sessionIdMiddleware.getSessionId, userController.logout)
    return userRouter
}

export default {
    userRouter
}